docs/design-guides/architecture-overview.md

```diff
@@ -0,0 +1,49 @@
+3. Azure Communication Identity service which generates ACS Identity and Tokens. The interaction with ACS Identity service happens within [ACS sdk](https://docs.microsoft.com/en-us/azure/communication-services/concepts/sdk-options#sdks) used within the solution. 
+> Note: The `api/token/teams` endpoint does not leverage #2, as the M365 Azure Active Directory Identity is mapped to user's Teams Identity within ACS Identity Service.
+
+### Motivation for leveraging Graph Open Extensions for Identity Mapping
+The primary motivation comes from the fact that the Azure Communication Identity for the user could be co-located with rest of the information for the Active Active Directory user. And as that would also prevent the usage of additional storage for storing ACS Identity mappings corresponding to Azure Active Directory user Id.
+
+## Limitations
+- The open extensions on Azure Active Directory can only be supported upto 2 per user instance as suggested [here](https://docs.microsoft.com/en-us/graph/extensibility-overview#open-extension-limits). Hence supporting multi tenancy could be a challenge as different extensions might be desired for different tenants.
+- There is Graph open extensions rate limit of 455 requests per 10 seconds as suggested [here](https://docs.microsoft.com/en-us/graph/throttling#open-and-schema-extensions-service-limits)
+- The sample solution leveraging Graph open extensions might not be suitable for the scenarios where the Azure Active Directory instance is used for 3rd party application sign in as with [delegated permissions granted over Graph Api](https://docs.microsoft.com/en-us/graph/auth/auth-concepts#delegated-and-application-permissions). The 3rd Party application in this case with right delegated permissions would be able to access the open extension data of the user.
+> Note: From data privacy perspective, the ACS Identity is categorized as [EUPI data](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/properties/devenv-dataclassification-property#property-value)
+
+## Guidance
+1. To support the multi environment scenario within Graph Open Extensions using the same Azure Active Directory user instance, you could extend the [IdentityMappingModel class](https://github.com/Azure-Samples/communication-services-authentication-hero-csharp/blob/main/src/Models/IdentityMappingModel.cs) to handle multiple environments.
+2. If Graph open extensions does not meet your requirement for storage of Identity Mappings, you can consider storing them in databases as [CosmoDB](https://docs.microsoft.com/en-us/azure/cosmos-db/) or [Azure Tables](https://docs.microsoft.com/en-us/azure/storage/tables/) for an example. And you would need to update the write and read operations in [GraphService.cs class](https://github.com/Azure-Samples/communication-services-authentication-hero-csharp/blob/main/src/Services/GraphService.cs)
```

Comment:
This I think we need to re-word a little bit. This whole section talks about alternative approach to storing the mappings in some kind of a database. I feel like instead of a bullet point this deserves a separate section "Other Identify Mapping Approaches".

nits:
* "CosmosDB",
* "ACS" -> "Azure Communication Services"